CONTRIBUTING.adoc

```diff
@@ -193,6 +193,103 @@ Follow these steps to ensure that virtual box interfaces are unmanaged:
 
     $ sudo nmcli con reload
 
+=== Develop and test using a docker-in-docker cluster
+
+It's possible to run an OpenShift multinode cluster on a single host
+via docker-in-docker (dind).  Cluster creation is cheaper since each
+node is a container instead of a VM.  This was implemented primarily
+to support multinode network testing, but may prove useful for other
+use cases.
+
+To run a dind cluster in a VM, follow steps 1-3 of the Vagrant
+instructions and then execute the following:
+
+        $ export OPENSHIFT_DIND_DEV_CLUSTER=true
+        $ vagrant up
+        $ vagrant ssh
+        $ cd /vagrant
+        $ hack/dind-cluster.sh start
+
+These steps will create a new VM, update its packages, ensure that
+docker is installed, and launch a new OpenShift dind cluster.  Package
+installation and the building of the dind image will take some time,
+but are a one-time cost.  Subsequent cluster launches should be
+considerably faster.
```

Comment:
Yeah, maybe split it up into two parts, with everything up to "vagrant up" being explained as one-time setup, and then ssh+cd+start as the iteratable part.